server/src/main/java/org/elasticsearch/action/admin/cluster/stats/TransportClusterStatsAction.java

```diff
@@ -77,14 +88,43 @@ protected ClusterStatsResponse newResponse(ClusterStatsRequest request,
                                                List<ClusterStatsNodeResponse> responses, List<FailedNodeException> failures) {
         assert Transports.assertNotTransportThread("Constructor of ClusterStatsResponse runs expensive computations on mappings found in" +
                 " the cluster state that are too slow for a transport thread");
-        ClusterState state = clusterService.state();
+        final ClusterState state = clusterService.state();
+        final Metadata metadata = state.metadata();
+        MappingStats mappingStats = null;
+        AnalysisStats analysisStats = null;
```

Comment:
Just a drive-by comment (I'll leave the full review to Dan), but can you rename these variables? I think shadowing class variables (outside of constructors) can lead to bugs for people who reference it not realizing which one they're getting